ycmd.el

```diff
@@ -551,7 +551,8 @@ and `delete-process'.")
     (after-change-functions           . ycmd--on-change)
     (window-configuration-change-hook . ycmd--on-window-configuration-change)
     (kill-buffer-hook                 . ycmd--teardown)
-    (before-revert-hook               . ycmd--teardown))
+    (before-revert-hook               . ycmd--teardown)
+    (hack-local-variables-hook        . ycmd--on-open))
```

Comment:
Why `hack-local-variables-hook`? Wouldn't it be enough to cann `ycmd--on-open` when the mode is activated? Basically calling the function in the body of the minor mode definition? Or in the minor mode `after-hook` before calling `(ycmd--comditional-parse 'mode-enabled)`?